src/concepts/plugins/plugin_basics.md

```diff
@@ -0,0 +1,430 @@
+- Handle authentication, authorization and data calls.
+
+### Full Stack Plugins
+
+_For advanced documentation on Full Stack Plugins - click [here](/concepts/plugins/fullstack.md)._
+
+Full Stack Plugins combine both the frontend and backend functionality into a single plugin, allowing developers to extend and customize both the client-side and server-side ascpects of the Caido application. These Plugins utilize the entire technology stack - from the user interface to the underlying server infrastructure.
+
+With full stack Plugins you can:
+
+- Facilitate the exchange of data and functionality between both the client and server.
+- Manage the flow of data between the frontend and backend.
+
+## SDKs
+
+An **SDK** (Software Development Kit) is an interface that provides access to various services and functionalities. Caido provides different types of SDKs dependent on the type of the Plugin being developed.
+
+A general description of the different SDK types are listed below:
+
+### Frontend SDKs
+
+_For advanced documentation on Caido's Frontend Plugin SDK - click [here](/concepts/plugins/frontend_sdk.md)._
+
+These development kits are specifically designed to help developers build plugins that run client-side. The tools, libraries and APIs that facilitate interaction with the user interface, handle client-side logic and communicate with the backend or APIs are included in the kit. Often, frontend SDKs include features such as user interface components, data manipulation libraries and integration with external services.
+
+### Backend SDKs
+
+_For advanced documentation on Caido's Backend Plugin SDK - click [here](/concepts/plugins/backend_sdk.md)._
+
+These SDKs assist developers in building server-side plugins. They also provide the tools, libraries and APIs to perform operations. Server-side operations include handling incoming requests from clients, managing databases, implementing business logic and integrating with external services. Scalability, security and performance optimization are all aspects given focus to server-side development.
+
+### Full Stack SDKs
+
+_For advanced documentation on Caido's Full Stack Plugin SDK - click [here](/concepts/plugins/fullstack_sdk.md)._
+
+These kits combine the frontend and backend functionality into a single package. Utilizing a full stack SDK enables developers to build Plugins that span both the client-side and server-side. These SDKs often include additional features for managing data flow between the two sides, handling authentication/authorization and providing seamless integration.
+
+## Conceptualizing Plugins
+
+Think of Plugins in Caido as extensive [Workflows](/concepts/essentials/workflows.md). While they both provide task automation - Plugin development offers a greater level of complexity.
+
+## Plugin Development
+
+Plugins consist of packages - meaning a collection of directories and files.
+
+### Contents of the StarterKit Repository
```

Comment:
That should not go in here, that will be specific to each plugin type

Reply:
So the comment on lines 106 to 129 - all of those files won't be included in every repo type?